### ISSUE TITLE
Specify IPFS node on the network instead of running localy

### ISSUE THREAD

**Issue by IngwiePhoenix** [ipfs/ipfs-desktop#2441]
**Is your feature request related to a problem? Please describe.**
I couldn't find it in the app itself, so I thought I'd take it here.

I run an IPFS node on the network which is also always reachable through VPN. Since this also has the most storage available, I would love to just use IPFS Desktop as a client, rather than an individual node.

Is there a good way to do that? If not, this is my feature request. :)

**Describe the solution you'd like**
Aside from selecting a custom IPFS binary, also allow selecting if a local or remote node should be used.

**Describe alternatives you've considered**
Allow to set a config option to do this instead.

**Additional context**
None I could think of, sorry.

**Comment by IngwiePhoenix** [comment]
Hello @whizzkid !

I do use the WebUI to see the settings and adjust the configuration, and for that it is very helpful. However, I really like the convenience of having context menu entries in Windows Explorer to quickly add files to my IPFS node, which I can then use to share files with my friends. The only alternative I have found so far is using ShareX and creating a custom integration that would `POST` the file to the API endpoint (`/add` I believe it was? I haven't set it up, I just know it works in theory).

Since I move between my Desktop, Laptop and my Phone a lot, my idea was to use a central IPFS node as the main "gateway" or "service" to share my files with. It is super convenient, after all.

For now, I have just uploaded files via the WebUI manually for the time being. But in the future, I would like to be able to just use the context menu entry instead to do so.

Thanks!

**Comment by whizzkid** [comment]
You probably meant @whizzzkid :)

**Comment by IngwiePhoenix** [comment]
> You probably meant @whizzzkid :)

damnit, derp! XD Sorry for the out of nowhere ping! ^^;

**Comment by IngwiePhoenix** [comment]
> In the meantime to support this, we can explore the possibility of turning off daemon and instead request the path to the `node`.

Those suggestions sound very much like what I am looking for! :)

One question, just so I got things sorted in my head: When you say API, you mean the one running on `$ip:5001`, yes? Then, where exactly is the RPC mounted at?

Thank you for your insight!

**Comment by IngwiePhoenix** [comment]
I see! Well, this shouldn't be a big problem for me; I have a good UFW setup :) Never seen that RPC before though, going to have to look into that!

**Comment by SgtPooki** [comment]
If you set your gateway api url (your single ipfs node endpoint) in settings, it should try to use that one and not boot up kubo behind the scenes.

E.g. i can run different versions of kubo manually in my terminal (RC and others) and put the url to that one in the settings to use that instead of starting up the kubo version bundled with desktop

**Comment by BigLep** [comment]
General thought: can SSH tunneling be used for this?  It would prevent needing to write extra code and keep the security property of not exposing one's Kubo RPC endpoint publicly.

**Comment by BigLep** [comment]
I adjusted my message to discuss the security property of not "exposing the Kubo RPC endpoint beyond localhost".  

Realistically unless there is a lot of demand here, I don't imagine this feature would get picked up by a core maintainer, especially if there is a way for a user to unblock themselves with an SSH tunnel (which is why I wanted to confirm that an SSH tunnel would meet the need here).

**Comment by lidel** [comment]
> I run an IPFS node on the network which is also always reachable through VPN. Since this also has the most storage available, I would love to just use IPFS Desktop as a client, rather than an individual node.


IPFS Desktop is an app for running and controlling a localhost Kubo node. Adding first-class remote node management is out of scope, so closing this.

That said, there are two ways to work with remote nodes:

- For nodes in the same LAN, or tunneled over SSH, load WebUI in regular web browser (e.g. `http://192.168.1.100:5001/webui`)
- IPFS Desktop: yo ucould try a manual workaround using the standard `$IPFS_PATH/api` file. **This is unsupported -- you do it at your own risk.**

### Manual workaround

  1. Make sure IPFS Desktop is not running
  2. Ensure the repo directory exists (default `~/.ipfs`)
  3. Remove the `config` file from that directory (its absence tells Desktop to skip starting a local daemon)
  4. Create an `api` file with the multiaddr of the remote node's RPC API:
     `echo "/ip4/192.168.1.100/tcp/5001" > ~/.ipfs/api`
  5. Start IPFS Desktop -- it will connect to the remote API instead of spawning a local daemon

  To revert, stop Desktop, remove the `api` file, and let it reinitialize on next start.

  ### Obligatory Security Note

  **NEVER expose the Kubo RPC API to the public internet.** It grants full control over the node. See the [RPC API docs](https://docs.ipfs.tech/reference/kubo/rpc/).

  If you must access a remote node's API, set up [`API.Authorizations`](https://github.com/ipfs/kubo/blob/master/docs/config.md#apiauthorizations) and use   `--api-auth` when connecting via CLI.

  For accessing nodes behind NATs, look into SSH or [p2p tunnels](https://github.com/ipfs/kubo/blob/master/docs/p2p-tunnels.md) (or both (ssh over p2p)).

  ### Recommended alternative

  The Kubo `ipfs` CLI is a better fit for working with remote nodes -- it has native `--api` and `--api-auth` support:

  `ipfs --api /ip4/192.168.1.100/tcp/5001 --api-auth [secret]  id`
